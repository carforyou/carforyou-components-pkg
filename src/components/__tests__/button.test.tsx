import React from "react"
import { render, fireEvent, cleanup } from "@testing-library/react"

import Button from "../button"

describe("<Button>", () => {
  beforeEach(cleanup)

  it("renders button label", () => {
    const { getByText } = render(<Button>Label</Button>)
    getByText("Label")
  })

  it("triggers onClick", () => {
    const onClick = jest.fn()
    const { getByText } = render(<Button onClick={onClick}>Label</Button>)

    fireEvent.click(getByText("Label"))
    expect(onClick).toHaveBeenCalled()
  })

  it("renders default button correctly", () => {
    const { container } = render(<Button>Label</Button>)
    expect(container).toMatchSnapshot()
  })

  describe("button variations", () => {
    it("renders small button", () => {
      const { container } = render(<Button small>Label</Button>)
      expect(container).toMatchSnapshot()
    })

    it("renders teal button", () => {
      const { container } = render(<Button teal>Label</Button>)
      expect(container).toMatchSnapshot()
    })

    it("renders disabled button", () => {
      const { container } = render(<Button disabled>Label</Button>)
      expect(container).toMatchSnapshot()
    })

    it("renders small disabled button", () => {
      const { container } = render(
        <Button disabled small>
          Label
        </Button>
      )
      expect(container).toMatchSnapshot()
    })

    it("renders small teal button", () => {
      const { container } = render(
        <Button small teal>
          Label
        </Button>
      )
      expect(container).toMatchSnapshot()
    })

    it("renders white Teal button", () => {
      const { container } = render(<Button tealBorder>Label</Button>)
      expect(container).toMatchSnapshot()
    })

    it("renders white Teal small button", () => {
      const { container } = render(
        <Button tealBorder small>
          Label
        </Button>
      )
      expect(container).toMatchSnapshot()
    })
  })

  it("has correct padding without the custom renderer", () => {
    const { getByText } = render(<Button>Label</Button>)
    const button = getByText("Label")

    expect(button.classList).toContain("py-16")
    expect(button.classList).toContain("px-10")
  })

  it("sets correct padding within the link", () => {
    const { getByText } = render(
      <Button>
        <a>Label</a>
      </Button>
    )

    const link = getByText("Label")
    expect(link.classList).toContain("py-16")
    expect(link.classList).toContain("px-10")
  })
})
