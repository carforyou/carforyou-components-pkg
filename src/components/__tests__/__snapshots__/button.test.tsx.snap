// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button> button variations renders disabled button 1`] = `
<div>
  <button
    class="flex w-12/12 justify-center items-center text-white leading-xs transition-2 cursor-pointer font-bold text-base rounded focus:outline-none px-10 py-16 bg-salmon hover:bg-salmon-dark focus:bg-salmon cursor-not-allowed bg-grey-3 hover:bg-grey-3"
    disabled=""
    type="button"
  >
    Label
  </button>
</div>
`;

exports[`<Button> button variations renders small button 1`] = `
<div>
  <button
    class="flex w-12/12 justify-center items-center text-white leading-xs transition-2 cursor-pointer font-bold text-base rounded focus:outline-none px-10 py-8 bg-salmon hover:bg-salmon-dark focus:bg-salmon"
    type="button"
  >
    Label
  </button>
</div>
`;

exports[`<Button> button variations renders small disabled button 1`] = `
<div>
  <button
    class="flex w-12/12 justify-center items-center text-white leading-xs transition-2 cursor-pointer font-bold text-base rounded focus:outline-none px-10 py-8 bg-salmon hover:bg-salmon-dark focus:bg-salmon cursor-not-allowed bg-grey-3 hover:bg-grey-3"
    disabled=""
    type="button"
  >
    Label
  </button>
</div>
`;

exports[`<Button> button variations renders small teal button 1`] = `
<div>
  <button
    class="flex w-12/12 justify-center items-center text-white leading-xs transition-2 cursor-pointer font-bold text-base rounded focus:outline-none px-10 py-8 bg-teal hover:bg-teal-dark focus:bg-teal"
    type="button"
  >
    Label
  </button>
</div>
`;

exports[`<Button> button variations renders teal button 1`] = `
<div>
  <button
    class="flex w-12/12 justify-center items-center text-white leading-xs transition-2 cursor-pointer font-bold text-base rounded focus:outline-none px-10 py-16 bg-teal hover:bg-teal-dark focus:bg-teal"
    type="button"
  >
    Label
  </button>
</div>
`;

exports[`<Button> button variations renders white Teal button 1`] = `
<div>
  <button
    class="flex w-12/12 justify-center items-center text-white leading-xs transition-2 cursor-pointer font-bold text-base rounded focus:outline-none px-10 py-16 bg-white border border-teal text-teal hover:opacity-60"
    type="button"
  >
    Label
  </button>
</div>
`;

exports[`<Button> button variations renders white Teal small button 1`] = `
<div>
  <button
    class="flex w-12/12 justify-center items-center text-white leading-xs transition-2 cursor-pointer font-bold text-base rounded focus:outline-none px-10 py-8 bg-white border border-teal text-teal hover:opacity-60"
    type="button"
  >
    Label
  </button>
</div>
`;

exports[`<Button> renders default button correctly 1`] = `
<div>
  <button
    class="flex w-12/12 justify-center items-center text-white leading-xs transition-2 cursor-pointer font-bold text-base rounded focus:outline-none px-10 py-16 bg-salmon hover:bg-salmon-dark focus:bg-salmon"
    type="button"
  >
    Label
  </button>
</div>
`;

exports[`<Button> renders icon, size, paddings and text 1`] = `
<div>
  <button
    class="flex w-12/12 justify-center items-center text-white leading-xs transition-2 cursor-pointer font-bold text-base rounded focus:outline-none px-10 py-10 bg-salmon hover:bg-salmon-dark focus:bg-salmon"
    type="button"
  >
    <svg
      height="32"
      width="32"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M13.454 9c.048.065.124.174.237.338.164.238.291.448.383.638.13.268.199.55.199.84 0 .508-.25.945-.68 1.398a7.924 7.924 0 0 1-1.008.894 7.555 7.555 0 0 0-.823.71l.026.059c.016.032.07.125.155.262.134.217.177.287.187.31.676 1.219 1.444 2.257 2.304 3.117.86.86 1.898 1.628 3.079 2.284.06.03.131.073.348.207.137.084.23.14.262.155a.979.979 0 0 0 .058.026 7.555 7.555 0 0 0 .71-.823 7.96 7.96 0 0 1 .895-1.008c.453-.43.89-.68 1.399-.68.289 0 .57.07.84.2.19.09.4.218.637.382.164.113.273.19.339.237.217.13.45.264.697.403.266.15.562.313.89.49.339.184.602.33.737.41.943.472 1.401.792 1.552 1.254.091.214.123.436.123.692 0 .343-.057.747-.167 1.226a6.61 6.61 0 0 1-.351 1.136c-.308.732-1.035 1.363-2.178 1.998-1.032.559-2.075.845-3.12.845-.3 0-.594-.02-.881-.059a7.117 7.117 0 0 1-.952-.206 12 12 0 0 1-.738-.226 56.434 56.434 0 0 1-.805-.297c-.365-.137-.593-.22-.681-.25a13.85 13.85 0 0 1-2.675-1.27c-1.283-.792-2.594-1.86-3.939-3.205C9.17 20.142 8.1 18.83 7.31 17.552a13.905 13.905 0 0 1-1.279-2.699 31.836 31.836 0 0 0-.243-.661 56.434 56.434 0 0 1-.297-.805 12.366 12.366 0 0 1-.226-.738 7.117 7.117 0 0 1-.206-.952A6.479 6.479 0 0 1 5 10.815c0-1.044.286-2.087.85-3.127.63-1.135 1.261-1.862 1.977-2.163.303-.133.684-.25 1.152-.358.48-.11.883-.167 1.226-.167.256 0 .478.032.614.094.54.18.86.638 1.302 1.527.11.19.256.452.44.791.177.328.34.624.49.89.139.247.273.48.403.697z"
        fill="none"
        stroke="#FFF"
        stroke-width="2"
      />
    </svg>
    <span
      class="pl-8 pr-16"
    >
      Label
    </span>
  </button>
</div>
`;
